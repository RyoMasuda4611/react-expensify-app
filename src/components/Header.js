import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/'>Go Home</NavLink>
        <NavLink to='/create'>Go AddExpensePage</NavLink>
        <NavLink to='/edit'>Go EditExpensePage</NavLink>
        <NavLink to='/help'>Go HelpPage</NavLink>        
    </header>
);

export default Header;