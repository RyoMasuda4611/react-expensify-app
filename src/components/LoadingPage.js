import React from 'react';

const LoadingPage = () => (
  <div className="loader">
      <img className="loader__images" src="/images/loader.gif"/>
  </div>
);

export default LoadingPage;
