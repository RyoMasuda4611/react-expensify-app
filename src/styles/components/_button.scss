.button {
  background: $blue;
  color: white;
  border: none;
  display: inline-block; //ブロック要素として扱う
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
  line-height: 1;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}