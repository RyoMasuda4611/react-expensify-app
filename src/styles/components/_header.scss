.header {
  background: $dark-blue;
}

.header__content {
    align-items: center;
    justify-content: space-between;
    display: flex;
    padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none; // invalidate default effection
  h1 {
    margin: 0;
  }
}
