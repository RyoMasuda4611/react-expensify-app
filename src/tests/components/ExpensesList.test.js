import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render Expense List correctly', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Expense List correctly without expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
