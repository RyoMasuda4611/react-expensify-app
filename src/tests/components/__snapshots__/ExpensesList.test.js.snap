// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense List correctly 1`] = `
<div>
  <ExpenseItemList
    amount={10}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseItemList
    amount={1000}
    createdAt={-345600000}
    description="Credit Card"
    id="2"
    key="2"
    note=""
  />
  <ExpenseItemList
    amount={1000000}
    createdAt={345600000}
    description="Rent"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render Expense List correctly without expenses 1`] = `
<div>
  <p>
    No Expense
  </p>
</div>
`;
